### PR TITLE
Add functions to control GPIOs

### LINKED ISSUE #13: Add GPIO functions
The TMF8801 has two gpios: GPIO0 and GPIO1.
please add some functions to control these pins.

### LINKED ISSUE #39: crashes when measurement_period_ms = 0
### :bug: Bug Report

<p align="center">
  <a href="https://github.com/107-systems/107-Arduino-Viper"></a>
</p>

**Describe the bug**

crashes when measurement_period_ms = 0 for example for single measurement

**Minimum example code to replicate the bug**
```C++
tmf8801.begin(0);
```

**Minimum configuration to replicate the bug**
Please provide the configuration data used to build Snowfox:
* Build Toolchain: (e.g. Arduino IDE 1.8.13, arduino-cli 0.13.0)
* Arduino Core: (e.g. ArduinoCore-samd 1.8.9)
* Library Version (of all involved libraries): (e.g. 107-Arduino-UAVCAN 1.1.0, 107-Arduino-MCP2515 1.2.0)

### PR DESCRIPTION
**Pull request by generationmake**
add functions to control GPIOs
set GPIO0 or GPIO1 to output and level 

This fixes #40.

### CODE REVIEW

File: src/107-Arduino-TMF8801.h [hunk 1]
```diff
@@ -75,6 +77,8 @@ class ArduinoTMF8801 : public LengthSensorBase
   TMF8801::AlgoState const & _algo_state;
   unit::Length _distance;
 
+  uint8_t gpio_control;
```
Comment (aentinger) [inline src/107-Arduino-TMF8801.h]:
```suggestion
  uint8_t _gpio_control;
```
Typically member variables are prefixed with `_`.

File: src/107-Arduino-TMF8801.h [hunk 2]
```diff
@@ -55,6 +55,8 @@ class ArduinoTMF8801 : public LengthSensorBase
 
   bool begin(uint8_t const measurement_period_ms);
 
+  void set_gpio(uint8_t num, uint8_t val);
```
Comment (aentinger) [inline src/107-Arduino-TMF8801.h]:
```suggestion
  void set_gpio(uint8_t const num, uint8_t const val);
```
Use const-correctness ;)
Better: Split API in 2 functions and use enum classes, i.e.
```C++
enum class GPIO : uint8_t
{
  GPIO0 = 0,
  GPIO1 = 1,
};
void set_gpio(GPIO const gpio);
void clr_gpio(GPIO const gpio);
```

File: src/107-Arduino-TMF8801.cpp [hunk 3]
```diff
@@ -80,14 +80,18 @@ bool ArduinoTMF8801::begin(uint8_t const measurement_period_ms)
   _api.clearInterrupt (TMF8801::InterruptSource::ObjectDectectionAvailable);
   _api.enableInterrupt(TMF8801::InterruptSource::ObjectDectectionAvailable);
 
+  /* set gpio_control to Input for GPIO0 and GPIO1
+   */
+  gpio_control=0x00;
```
Comment (aentinger) [inline src/107-Arduino-TMF8801.cpp]: It's better to initialize member variables with default values in the Constructor, please move initialisation up there.

**Review by aentinger — CHANGES_REQUESTED** [review]
Good work :+1: How about adding a simple example demonstrating GPIO control?

**Comment by generationmake** [comment]
Hi @aentinger 
Thanks for the input.
I will add and change this tonight. I also have an example ready.

File: src/107-Arduino-TMF8801.cpp (hunk 3 above)
Comment (generationmake) [reply src/107-Arduino-TMF8801.cpp]: should be fixed now.

**Review by generationmake — COMMENTED** [review]
should be fixed now

**Review by aentinger — APPROVED** [review]
LGTM :+1:

**Pull request merged**